pkg/ccl/sqlproxyccl/balancer/conn_tracker.go

```diff
@@ -140,6 +230,9 @@ func (e *tenantEntry) addHandle(handle ConnectionHandle) bool {
 		return false
 	}
 	e.mu.conns[handle] = struct{}{}
+
+	// New connections are always active.
+	e.cache.updateActiveCount(handle.ServerRemoteAddr(), 1)
```

Comment:
There is still a risk of assigning all of the connections to a single sql server. The connection is added to the connection tracker after it is successfully proxied to the backend. That means there is a connection latency gap (~40ms) between assigning a connection and updating the connection counts. 40ms is significant, because connection attempts are not random. If a client with a connection pool starts up, a sql proxy would assign all of the pool's connections to the same sql server.

How large of a change would it be to add a connection to the connection tracker as soon as it has a backend assignment? There would still be a bit of a race, but it would be 1us-1ms and probably would not matter.